### ISSUE TITLE
[k8s] file mount fails when using job controller

### ISSUE THREAD

**Issue by asaiacai** [skypilot-org/skypilot#3530]
I had the following task and config which I executed on k8s with `sky jobs launch -n job5 sky-kueue-task.yaml`

` sky-kueue-task.yaml`
```
resources:
  cloud: kubernetes
  accelerators: H100:8
  labels:
    kueue.x-k8s.io/queue-name: user-queue-a

num_nodes: 2

file_mounts:
  ~/spin.py: ./spin.py

setup: |
  pip install torch

run: |
  python ~/spin.py
```
`~/.sky/config.yaml`
```
kubernetes:
  provision_timeout: -1  # Wait indefinitely in the queue to get scheduled on the k8s cluster
  remote_identity: SERVICE_ACCOUNT
```
This hangs indefinitely. When I log into the jobs controller, it looks like there's a 403 error on `skypilot-service-account` for creating a namespace during the FUSE device manager creation.

```
I 05-09 18:36:18 config.py:455] _configure_fuse_mounting: Setting up FUSE device manager.
E 05-09 18:36:18 provisioner.py:91] Failed to configure 'job5-2' on Kubernetes Region(name='kubernetes') (all zones) with the following error:
E 05-09 18:36:18 provisioner.py:91] kubernetes.client.exceptions.ApiException: (403)
E 05-09 18:36:18 provisioner.py:91] Reason: Forbidden
E 05-09 18:36:18 provisioner.py:91] HTTP response headers: HTTPHeaderDict({'Audit-Id': '994b9637-1caa-4fbd-ba89-b27de4fcfcd3', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'X-Kubernetes-Pf-Flowschema-Uid': 'aef310ce-b0dc-4769-9e55-1a2febf4a813', 'X-Kubernetes-Pf-Prioritylevel-Uid': 'c67a67d7-3d69-4dea-90b4-7b626853112c', 'Date': 'Thu, 09 May 2024 18:36:18 GMT', 'Content-Length': '311'})
E 05-09 18:36:18 provisioner.py:91] HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"namespaces is forbidden: User \"system:serviceaccount:default:skypilot-service-account\" cannot create resource \"namespaces\" in API group \"\" at the cluster scope","reason":"Forbidden","details":{"kind":"namespaces"},"code":403}
```

_Version & Commit info:_
* `sky -v`: 1.0.0.dev20240509
* `sky -c`: 07da93b3c3b87e53d0ebd7e033c9ce24992151fe

**Comment by romilbhardwaj** [comment]
Great catch @asaiacai! Should be fixed in #3532. Would you like to give it a go?

**Comment by asaiacai** [comment]
awesome, thanks for the quick fix!
